packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -711,78 +785,92 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
-                    );
-                };
-                const snapshot = await fetchSnapshot(
-                    this.snapshotUrl!,
-                    storageToken,
-                    id,
-                    this.fetchFullSnapshot,
-                    !!this.hostPolicy.sessionOptions?.forceAccessTokenViaAuthorizationHeader,
-                    this.logger,
-                    snapshotDownloader,
-                );
-                let treeId = "";
-                if (snapshot.snapshotTree) {
-                    assert(snapshot.snapshotTree.id !== undefined, 0x222 /* "Root tree should contain the id!!" */);
-                    treeId = snapshot.snapshotTree.id;
-                    this.setRootTree(treeId, snapshot.snapshotTree);
-                }
-                if (snapshot.blobs) {
-                    this.initBlobsCache(snapshot.blobs);
-                }
-                // If the version id doesn't match with the id of the tree, then use the id of first tree which in that case
-                // will be the actual id of tree to be fetched.
-                return this.commitCache.get(id) ?? this.commitCache.get(treeId);
-            });
-        }
+export class LocalOdspDocumentStorageService extends OdspDocumentStorageServiceBase {
+    constructor(
+        resolvedUrl: IOdspResolvedUrl,
+        _logger: ITelemetryLogger,
+        private readonly fluidFile: Uint8Array | string,
+    ) {
+        super(
+            resolvedUrl,
+        );
+    }
 
-        if (!tree) {
-            return null;
+    public async getVersions(blobid: string | null, count: number): Promise<api.IVersion[]> {
+        let snapshotContents: ISnapshotContents;
+
+        if (typeof this.fluidFile === "string") {
+            const content: IOdspSnapshot = JSON.parse(this.fluidFile);
+            snapshotContents = convertOdspSnapshotToSnapshotTreeAndBlobs(content);
+        } else {
+            snapshotContents = parseCompactSnapshotResponse(
+                new ReadBuffer(this.fluidFile));
         }
 
-        return tree;
+        const id = this.initializeFromCachedValue(snapshotContents);
+
+        // Look at fetchSnapshot.ts > downloadSnapshot(...)
+        return id ? [{ id, treeId: undefined! }] : [];
     }
 
-    private combineProtocolAndAppSnapshotTree(
-        hierarchicalAppTree: api.ISnapshotTree,
-        hierarchicalProtocolTree: api.ISnapshotTree,
-    ) {
-        const summarySnapshotTree: api.ISnapshotTree = {
-            blobs: {
-                ...hierarchicalAppTree.blobs,
-            },
-            trees: {
-                ...hierarchicalAppTree.trees,
-                // the app tree could have a .protocol
-                // in that case we want to server protocol to override it
-                ".protocol": hierarchicalProtocolTree,
-            },
-        };
+    protected async fetchTreeFromSnapshot(id: string): Promise<api.ISnapshotTree | undefined> {
+        return undefined;
```

Comment:
This method and fetchBlobFromStorage should log error to telemetry and throw exception